hooks/basic/collector.py

```diff
@@ -207,18 +207,76 @@ def on_context_changed(self, settings, item):
         """
 
         path = item.properties["path"]
-        is_sequence = item.properties["is_sequence"]
 
-        # get info for the item
-        item_info = self._get_item_info(settings, path, is_sequence)
-        work_path_template = item_info["work_path_template"]
+        item.properties["work_path_template"] = self._resolve_work_path_template(item.properties, path)
+
+        # Set the item's fields property
```

Comment:
The work_path_template is dependent on the context, so _get_item_info() needs to be run here when the context has been changed

Reply:
I can evaluate _get_item_info() in _resolve_work_path_template so that we get updated templates.